app/controllers/api/teams.php

```diff
@@ -343,11 +343,17 @@
             Query::limit(2000), // TODO fix members limit
         ]);
 
-        // TODO delete all members individually from the user object
         foreach ($memberships as $membership) {
             if (!$dbForProject->deleteDocument('memberships', $membership->getId())) {
                 throw new Exception(Exception::GENERAL_SERVER_ERROR, 'Failed to remove membership for team from DB');
             }
+
+            $user = $dbForProject->getDocument('users', $membership->getAttribute('userId'));
+            $user->setAttribute('memberships', array_values(array_filter(
+                $user->getAttribute('memberships', []),
+                fn($um) => $um['teamId'] !== $membership->getAttribute('teamId')
+            )));
+            $dbForProject->updateDocument('users', $user->getId(), $user);
```

Comment:
The big drawback about doing it in here is that the API can then take much longer to finish since it must iterate over every user, but the API should be as quick as possible.

Reply:
I'm gonna spend some time to see how and where the membership data is stored exactly. Will also look into background workers. Will leave a comment soon!

Reply:
Okay so based on what I see in mariadb, there is a `memberships` column in the `_1_users` table, but nothing gets stored in it even after I create a membership. As far as I can understand, when a membership is created, it does various checks and creates an entry in the `_1_memberships` table and that's it. There's no more entries in any other db as far as I can tell from the code and the db.

Going through the code in the console, it seems like the memberships page simply calls the `listMemberships` endpoint with user params. So the issue must be something to do with the frontend because the memberships get deleted from the `_1_memberships` table as expected, and memberships are displayed from a Teams API endpoint and not an Account API. I'll be reverting the code back to it's original state, and fixing the frontend issue. 

Does that sound good?

Reply:
Okay after some more digging, looks like I missed something super obvious. The sdk calls `users.listMemberships` which uses the `teams` table. Gonna look more into this and see what's up. Sorry for the notifs, and let me know if you want me to stop lmao